### ISSUE TITLE
Add ability to directly link to Mojang account on login

### ISSUE THREAD

**Issue by Phoenix616** [GeyserMC/Geyser#643]
**What feature do you want?**
This is a request of a functionality which allows directly linking your Bedrock account to your Mojang one instead of having to input your Mojang account's login data to play with your Mojang account or logging in with Floodgate.

Possible ways of doing this:
- Display an input field which accepts a linking code which you can get when logging in with your Mojang account and running a link command (easy method)
- Directly link your Xbox account to your Mojang account via Xbox's OAuth (this might be a bit more difficult to do and would require the Geyser software to host a website to get a callback from their API and might even require the server owner to setup their own xbox oauth app or a general geyser one with shared api keys)

Ideally this would be a new auth-type additionally to online, offline and floodgate. Due to the nature of it (requiring a link code from the server side) it might only be possible in the minecraft server/proxy parts or would have to be integrated in Floodgate (maybe it even fits better there, if so feel free to move the issue to that repo. If this would be better in Floodgate then no fake player should ever be visible to the actual server and plugins in order to not cause issues by non-mojang players).

Adding such a feature would enable Geyser to really be usable in front of public servers as it would neither require the user to trust the server by handing over their login data as well as make it compatible with every server plugin which might break due to how Floodgate works.

**Alternatives?**
A possible alternative would be to use Floodgate and connect players using it to a placeholder server where they can link their account, and only then they could switch to one of the other servers or do anything. This would make the setup way more complex than just having to install Geyser and linking.

**Comment by DoctorMacc** [comment]
It sounds like this already exists with Floodgate and `/linkaccount`, is it not?

**Comment by Heath123** [comment]
This should be an issue on Floodgate

**Comment by DoctorMacc** [comment]
Yes, this is something that would be handled with Floodgate. I'll transfer the issue.
